app/services/puppet_fact_parser.rb

```diff
@@ -140,6 +140,17 @@ def cores
     facts.dig('processors', 'count') || facts['processorcount']
   end
 
+  def total_disk_size
+    disks      = facts.select { |key, _| key.to_s =~ /blockdevice.*_size/ }
+    disks_size = 0
```

Comment:
This method must work also with structured facts, because discovery only supports Facter 2.X but core supports Facter 3.x and above. See above for an example:

```
  def ram
    if (value = facts.dig('memory', 'system', 'total_bytes'))
      value / 1.megabyte
    else
      facts['memorysize_mb']
    end
  end
```

Reply:
The discovery now uses facter 4 right?